crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -603,9 +604,17 @@ impl Actor for SenderAccount {
             .with_label_values(&[&sender_id.to_string()])
             .set(config.trigger_value as f64);
 
-        let sender_aggregator = HttpClientBuilder::default()
-            .request_timeout(config.rav_request_timeout)
-            .build(&sender_aggregator_endpoint)?;
+        let endpoint = Endpoint::new(sender_aggregator_endpoint.to_string())
+            .expect("Failed to create an endpoint for the sender aggregator")
+            .connect_timeout(config.rav_request_timeout);
+
+        let sender_aggregator = TapAggregatorClient::connect(endpoint.clone())
+            .await
+            .expect(&format!(
+                "Failed to connect to the TapAggregator endpoint '{}'",
+                endpoint.uri()
+            ))
```

Comment:
Using expecte is fair, but I'd rather just use `.with_context("Failed to connect to the TapAggregator endpoint")?` and let anyhow handle this up in the chain.

Reply:
It's interesting that gRPC now keeps a connection open instead of knowing that the aggregator is down only while trying to request.

This makes tests a little different since we need a gRPC server up for them to connect. I recommend taking a look in how we create tests that need aggregator and replicate the same.